Gemfile

```diff
@@ -10,6 +10,13 @@ gem "rails", rails_version == "main" ? {git: "https://github.com/rails/rails", r
 
 gem "rspec-rails", "~> 5"
 
+group :test do
+  # For testing UI interactions
+  gem "cuprite"
+  gem "puma"
+  gem "selenium-webdriver"
```

Comment:
Can you add coarse dependency pointers for these? Just to the major version is fine: `~> 1` etc.